implementations/elixir/ockam/ockam/lib/ockam/transport/tcp/wrapper.ex

```diff
@@ -0,0 +1,8 @@
+defmodule Ockam.Transport.TCP.Wrapper do
+  @moduledoc """
+  Behaviour definition for a TCP call wrapper.
+  Allows performing additional work around TCP calls, like
+  emitting telemetry events.
+  """
+  @callback wrap_tcp_call(fun :: function()) :: any()
```

Comment:
🤔 I wonder if the typespec should express that the return value is always equivalent to that of the inner gen_tcp send, and that the wrapper logic should only impart side effects?

Reply:
At least for now we want to make it relaxed and allow returning custom values.